### ISSUE TITLE
Error and crash using bind mounts

### ISSUE THREAD

**Issue by winstonhenke** [docker/for-win#6702]
- [x ] I have tried with the latest version of my channel (Stable or Edge)
  - [x ] I have uploaded Diagnostics
  - Diagnostics ID: 4C871A84-186D-4C6E-B611-CA37A0EAD207/20200513235407

### Expected behavior
Able to run a Docker container using bind mounts to my host PC

### Actual behavior
I receive errors in the logs and and the container crashes
### Information

  - Windows Version: 10.0.18363.778
  - Docker Desktop Version: 2.3.0.2 (45183)

### Steps to reproduce the behavior

Using this dockerfile
```
FROM mcr.microsoft.com/windows:10.0.18363.778
CMD ["powershell.exe"]
```

Build it using: `docker build -t mytest:latest .`

Run it using
```
docker run -it --isolation=hyperv `
--mount type=bind,source=C:/test1,target=C:/test1,readonly `
--mount type=bind,source=C:/test2,target=C:/test2 `
mytest:latest powershell
```
- Both C:/test1 and C:/test2 exist on my host

Causes the following in: `C:\Users\<me>\AppData\Local\Docker\log.txt`
```
[19:27:56.038][ApiProxy          ][Error  ] hostPathOfVolume C:/test1 failed, skipping volume
[19:27:56.039][ApiProxy          ][Error  ] hostPathOfVolume C:/test2 failed, skipping volume
```

Then it allows me to execute a couple commands before the container crashes with the following error
```
time="2020-05-13T19:34:53-05:00" level=error msg="Error waiting for container: failed to shutdown container: container ea956674c46c7e10f816db9f6b559ed905c1fd44dd68fc8b75656e07c94b1c67 encountered an error during hcsshim::System::Shutdown: failure in a Windows system call: The connection with the virtual machine or container was closed. (0xc037010a): subsequent terminate failed container ea956674c46c7e10f816db9f6b559ed905c1fd44dd68fc8b75656e07c94b1c67 encountered an error during hcsshim::System::waitBackground: failure in a Windows system call: The connection with the virtual machine or container was closed. (0xc037010a)"
```

One important note is that before the container crashes, if list the contents of one of the test directories from within the container, I am able to see the files that live on the host

edit1: removed `-ExecutionPolicy ByPass` from the run command in the STR

**Comment by winstonhenke** [comment]
Also when I run this container I have two error events logged to my event viewer with a docker source.
- I am logged into Docker Hub. Have tried logging in via the Docker Desktop app and the command line. Also I don't have issues pulling images from Docker Hub
- Also I verified using the `whoami` command that when I run and build the container it does so using `user manager\containeradministrator`

Event ID 11
```
exec's CreateProcess() failed [error=container e36b6674a1eaa3e4b520c86182954608754ccb5d1c954d159149701eef75dea7 encountered an error during hcsshim::System::CreateProcess: failure in a Windows system call: The user name or password is incorrect. (0x52e)
[Event Detail:  Provider: 00000000-0000-0000-0000-000000000000]
[Event Detail:  Provider: 00000000-0000-0000-0000-000000000000]
[Event Detail: onecore\vm\compute\management\orchestration\vmhostedcontainer\processmanagement.cpp(173)\vmcomputeagent.exe!00007FF65737AE77: (caller: 00007FF65732E4AB) Exception(16) tid(7a4) 8007052E The user name or password is incorrect.
    CallContext:[\Bridge_ProcessMessage\VmHostedContainer_ExecuteProcess] 
 Provider: 00000000-0000-0000-0000-000000000000] extra info: {"CommandLine":"cmd.exe /C \"ECHO 192.168.1.97    host.docker.internal \u003e\u003e %systemroot%\\system32\\drivers\\etc\\hosts \u0026 ECHO 192.168.1.97    gateway.docker.internal \u003e\u003e %systemroot%\\system32\\drivers\\etc\\hosts\"","User":"Administrator","WorkingDirectory":"/","CreateStdInPipe":true,"CreateStdOutPipe":true,"CreateStdErrPipe":true,"ConsoleSize":[0,0]} module=libcontainerd namespace=moby container=e36b6674a1eaa3e4b520c86182954608754ccb5d1c954d159149701eef75dea7 exec=810d6ee619ef04cbf63e785f4dda7af8c6e0cb238c866d595a9a867f51e82933]
```

Event ID 1
```
Error running exec 810d6ee619ef04cbf63e785f4dda7af8c6e0cb238c866d595a9a867f51e82933 in container: container e36b6674a1eaa3e4b520c86182954608754ccb5d1c954d159149701eef75dea7 encountered an error during hcsshim::System::CreateProcess: failure in a Windows system call: The user name or password is incorrect. (0x52e)
[Event Detail:  Provider: 00000000-0000-0000-0000-000000000000]
[Event Detail:  Provider: 00000000-0000-0000-0000-000000000000]
[Event Detail: onecore\vm\compute\management\orchestration\vmhostedcontainer\processmanagement.cpp(173)\vmcomputeagent.exe!00007FF65737AE77: (caller: 00007FF65732E4AB) Exception(16) tid(7a4) 8007052E The user name or password is incorrect.
    CallContext:[\Bridge_ProcessMessage\VmHostedContainer_ExecuteProcess] 
 Provider: 00000000-0000-0000-0000-000000000000] extra info: {"CommandLine":"cmd.exe /C \"ECHO 192.168.1.97    host.docker.internal \u003e\u003e %systemroot%\\system32\\drivers\\etc\\hosts \u0026 ECHO 192.168.1.97    gateway.docker.internal \u003e\u003e %systemroot%\\system32\\drivers\\etc\\hosts\"","User":"Administrator","WorkingDirectory":"/","CreateStdInPipe":true,"CreateStdOutPipe":true,"CreateStdErrPipe":true,"ConsoleSize":[0,0]}
```

**Comment by winstonhenke** [comment]
The number of mount points seems to impact how quickly the container crashes as well. The more mount points I add the quicker it crashes.
- 1-2 mount points it seems to not crash for a while
- 5-7 mount points and it crashes right away
- But the errors in the log and the event viewer happen with any number of mount points

The reason for so many mount points is that I'm working through dockerizing some old legacy stuff that is spread across multiple sources so until I can slim down those sources I was just using mount points to speed up development rather than trying pulling in multiple very large sources

**Comment by djg2002** [comment]
Maybe related to https://github.com/docker/for-win/issues/6700

**Comment by winstonhenke** [comment]
@djg2002 looks very similar. Only difference is I was having this issue before upgrading to v2.3.0.2 as well but looks like they have a fix in for this
- <https://github.com/docker/for-win/issues/6700#issuecomment-629133176>

**Comment by winstonhenke** [comment]
Might be fixed with v2.3.0.3
- https://docs.docker.com/docker-for-windows/release-notes/#docker-desktop-community-2303
